prow/plugins/trigger/generic-comment_test.go

```diff
@@ -417,7 +430,35 @@ func TestHandleGenericComment(t *testing.T) {
 					},
 				},
 			},
-			ShouldBuild: false,
+			ShouldBuild:  false,
+			ShouldReport: true,
+		},
+		{
+			name:   "Retest of run_if_changed job that failed. Changes do not require the job. Skipped statuses elided.",
```

Comment:
`/retest` should retrigger any ProwJob that has failed on the PR, not just required jobs.
If I run a ProwJob on my PR and it fails, I would expect `/retest` to rerun it whether it was a `run_if_changed` job, a manual trigger only job, or an automatically run job.

Reply:
This is a test that predates this change, all I did was dupe it and add the eliding to show that we skip the context correctly

Reply:
The eliding behavior is what is incorrect. If we don't post a skipped status or rerun the job, the context will still block the PR from merging. I think rerunning the job is the correct behavior in this case, but skipping may also be acceptable. We need to do something to fix the red status context in this case though.

Reply:
Isn't the correct approach here to use `/skip`?

Reply:
If we are set to elide, we're in the New Age of Prow Triggering and use `/skip`, that is

Reply:
Wait actually I thought we agreed that `/retest` forces a retest regardless of matching if it's failed. WTF. Let me look

Reply:
SGTM

Reply:
Actually, I'm still not convinced that we shouldn't be retriggering the job in this case, but this is a small edge case we can correct later. I want to get this PR in so Prow can be deployed again.